src/basic/ether-addr-util.c

```diff
@@ -15,6 +15,11 @@ char* hw_addr_to_string(const struct hw_addr_data *addr, char buffer[HW_ADDR_TO_
         assert(buffer);
         assert(addr->length <= HW_ADDR_MAX_SIZE);
 
+        if (addr->length == 0) {
+                buffer[0] = '\0';
+                return buffer;
+        }
+
         for (size_t i = 0; i < addr->length; i++) {
                 sprintf(&buffer[3*i], "%02"PRIx8, addr->bytes[i]);
```

Comment:
actually we probably should not use sprintf() here at all, but do:

```
buffer[3*i] = hexchar(addr->bytes[i] >> 4);
buffer[3*i+1] = hexchar(addr->bytes[i] & 0xF);
```

And then after the loop finished do:

```c
buffer[addr->length] = 0;
```

That's simpler, speedier and has less conditionalizations.